horizomer/utils/tree.py

```diff
@@ -62,9 +47,43 @@ def order_nodes(tree, increase=True):
     return res
 
 
-if __name__ == "__main__":
-    main()
+# replace siblings using children would cause ordering problems
+# assign prev to None for non_tips() would ignore root
+def is_nodes_ordered(tree, increase = True):
+    """Returns 'True' if the tree is ordered.
+
+    Parameters
+    ----------
+    tree : skbio.TreeNode
+        tree to check ordering
+    increase : bool, optional
+        check if nodes in increasing (True) or decreasing (False) order
+
+    Returns
+    -------
+    bool
+        'True' if the tree is ordered
+    """
+    tree_copy = tree.copy()
+    for node in tree_copy.postorder():
+        if node.is_tip():
+            node.n = 1
+        else:
+            node.n = sum(x.n for x in node.children)
 
+    p = tree_copy.root()
+    prev = p.n
+    for node in tree_copy.levelorder():
+        s = [x for x in p.siblings()]
+        if node in s:
```

Comment:
Guess what. Querying a list (array) is slower than querying a dict or set (hash). This step could slow down the whole function. Maybe (just maybe), the following code would accelerate:
```
for sibling in p.siblings():
    if node is sibling:
        cur = node.n
        if ...
        else ...
        break
```